apps/session-manager/src/controllers/session.ts

```diff
@@ -88,3 +96,61 @@ export const getSessionState: RTE.ReaderTaskEither<
     },
     (err) => new Error(String(err)),
   );
+
+export type LogoutDependencies = {
+  lollipopService: {
+    deleteAssertionRefAssociation: (typeof LollipopService)["deleteAssertionRefAssociation"];
+  };
+} & DependencyOf<
+  ReturnType<(typeof LollipopService)["deleteAssertionRefAssociation"]>
+> &
+  RedisSessionStorageServiceDepencency &
+  WithUser &
+  WithExpressRequest;
+
+export const logout: RTE.ReaderTaskEither<
+  LogoutDependencies,
+  Error,
+  IResponseSuccessJson<SuccessResponse>
+> = (deps) =>
+  pipe(
+    // retrieve the assertionRef for the user
+    deps.redisSessionStorageService.getLollipopAssertionRefForUser({
+      ...deps,
+      fiscalCode: deps.user.fiscal_code,
+    }),
+    TE.chain(
+      flow(
+        O.map((assertionRef) =>
+          pipe(
+            deps.lollipopService.deleteAssertionRefAssociation(
+              deps.user.fiscal_code,
+              assertionRef,
+              "lollipop.error.logout",
+              "logout from lollipop session",
+            )(deps),
+            TE.chain(
+              TE.fromPredicate(
+                (result) => result === true,
+                () => new Error("Error revoking the AssertionRef"),
+              ),
+            ),
+          ),
+        ),
+        // continue if there's no assertionRef on redis
+        O.getOrElseW(() => TE.of(true)),
+      ),
+    ),
+    TE.chain((_) =>
+      pipe(
+        deps.redisSessionStorageService.deleteUser(deps.user)(deps),
+        TE.chain(
+          TE.fromPredicate(
+            (result) => result === true,
+            () => new Error("Error destroying the user session"),
+          ),
+        ),
+      ),
+    ),
+    TE.map((_) => ResponseSuccessJson({ message: "ok" })),
```

Comment:
in `io-backend` implementation we had a `log.error` at this point of the logic. should we add it here?